### ISSUE TITLE
Custom highlight function doesn't handle ${VAR}

### ISSUE THREAD

**Issue by michael0liver** [pngwn/MDsveX#211]
Here's an example with a shell script its quite common to use `${VAR}` syntax inside a bash/shell script e.g:
````
```bash
VAR="MY_VAR"
echo "${MY_VAR}"
```
````

It looks like this conflicts with MDsveX/Svelte template syntax and never loads for me. This also affects javascript code blocks. Maybe making a note in the docs would make sense in https://github.com/pngwn/MDsveX/issues/210.

@pngwn noted a workaround by using a custom escaping function:
```js
const shiki = require("shiki");

const escape_svelty = (str) =>
  str
    .replace(
      /[{}`]/g,
      (c) => ({ "{": "&#123;", "}": "&#125;", "`": "&#96;" }[c])
    )
    .replace(/\\([trn])/g, "&#92;$1");

async function highlighter(code, lang) {
  const highlighter = await shiki.getHighlighter({ theme: "github-dark" });
  return escape_svelty(highlighter.codeToHtml(code, lang || "text"));
}

module.exports = {
  extensions: [".svx", ".md"],
  highlight: {
    highlighter: highlighter,
  },
};

```

_Originally posted by @michael0liver in https://github.com/pngwn/MDsveX/issues/205#issuecomment-803667947_

**Comment by michael0liver** [comment]
Related / mostly the same issue as https://github.com/pngwn/MDsveX/issues/117 after looking.

**Comment by pngwn** [comment]
Thank you for opening this issue.

I think the reason past me decide against this was to allow flexibility for users, in case they _wanted_ svelte to interpret the curlies found within code blocks for whatever reason but thinking about it there is a way to get both if we allow users to pass a flag somehow, `escape` which would signal whether or not try escaping the transformed contents or not. This would provide an escape hatch in the rare cases that it is needed.

**Comment by michael0liver** [comment]
That like a sensible idea. An `escape` key in the config which defaults to `true`.

**Comment by pngwn** [comment]
I don't think an `escape` prop will work because we don't want to escape all of the code that is returned from the custom highlight function.

In this case:

```js
const code = `<pre>${highlight(code)}</pre>`
```

We want to keep the `pre` tag intact and unescaped (we _want_ them to be interpreted as HTML). I think the solution to this is to export a utility function that people can use to escape the svelte syntax.

**Comment by pngwn** [comment]
Released in 0.9.1.
